tests/lib/hardware/libvirt.py

```diff
@@ -167,85 +176,129 @@ def _cloud_init_seed_create(self):
+                    <console type='pty'>
+                        <target type='serial' port='0'/>
+                    </console>
+                    <channel type='spicevmc'>
+                        <target type='virtio' name='com.redhat.spice.0'/>
+                        <address type='virtio-serial' controller='0' bus='0'
+                                 port='1'/>
+                    </channel>
+                    <input type='mouse' bus='ps2'/>
+                    <input type='keyboard' bus='ps2'/>
+                    <graphics type='spice' autoport='yes'/>
+                    <video>
+                        <model type='vga'/>
+                    </video>
+                    <redirdev bus='usb' type='spicevmc'></redirdev>
+                    <memballoon model='virtio'>
+                        <address type='pci' domain='0x0000' bus='0x00'
+                                 slot='0x06' function='0x0'/>
+                    </memballoon>
+                </devices>
+            </domain>
         """ % {
             "domain_name": domain_name, "image": image,
             "cloud_init_seed": cloud_init_seed,
             "network_name": network_name,
             "memory": memory
-        }
+        })
 
 
 class Hardware(HardwareBase):
     def __init__(self):
         super().__init__()
-        self._network = self.conn.networkLookupByName(
-            config.PROVIDER_LIBVIRT_NETWORK)
+        self._network = self.create_network()
         if not self._network:
             raise Exception(f'Can not get libvirt network '
-                            '{config.PROVIDER_LIBVIRT_NETWORK}')
+                            '{config.PROVIDER_LIBVIRT_NETWORK_RANGE}')
         logger.info(f"Got libvirt network {self._network.name()}")
+        self._image_path = self._get_image_path()
+
+    def _get_image_path(self):
+        if (config.PROVIDER_LIBVIRT_IMAGE.startswith("http://") or
+                config.PROVIDER_LIBVIRT_IMAGE.startswith("https://")):
+            logging.debug("Downloading image from URL")
+            download_location = os.path.join(
+                self.working_dir,
+                os.path.basename(config.PROVIDER_LIBVIRT_IMAGE)
+            )
+            wget.download(
+                config.PROVIDER_LIBVIRT_IMAGE,
+                download_location
+            )
+            return download_location
+        return config.PROVIDER_LIBVIRT_IMAGE
+
+    def create_network(self):
+        network = netaddr.IPNetwork(config.PROVIDER_LIBVIRT_NETWORK_RANGE)
+        network_name = "%s%s" % (config.CLUSTER_PREFIX, self._hardware_uuid)
+        host_ip = str(netaddr.IPAddress(network.first+1))
+        netmask = str(network.netmask)
+        dhcp_start = str(netaddr.IPAddress(network.first+2))
+        dhcp_end = str(netaddr.IPAddress(network.last-1))
+        xml = textwrap.dedent("""
+            <network>
+            <name>%(network_name)s</name>
+            <forward mode="nat"/>
```

Comment:
With that, we are tied to this type of network. But I guess that's fine for now?!